Data/ByteString/Lazy.hs

```diff
@@ -733,6 +804,23 @@ dropWhile f = dropWhile'
             n | n < S.length c -> Chunk (S.drop n c) cs
               | otherwise      -> dropWhile' cs
 
+-- | Similar to 'P.dropWhileEnd',
+-- drops the longest (possibly empty) suffix of elements
+-- satisfying the predicate and returns the remainder.
+--
+-- @'dropWhileEnd' p@ is equivalent to @'reverse' . 'dropWhile' p . 'reverse'@.
+--
+-- @since 0.11.1.1
+dropWhileEnd :: (Word8 -> Bool) -> ByteString -> ByteString
+dropWhileEnd f = dropWhileEnd'
+  where dropWhileEnd' Empty = Empty
+        dropWhileEnd' (Chunk c Empty) =
+            fromStrict (S.dropWhileEnd f c)
+        dropWhileEnd' (Chunk c bs) =
+            case dropWhileEnd' bs of
+                 Empty -> dropWhileEnd' (Chunk c Empty)
```

Comment:
This is a bit entangled, to be honest. If you put `fromStrict (S.dropWhileEnd f c)` instead of `dropWhileEnd' (Chunk c Empty)`, you'll be able to remove special handling of `Chunk c Empty` case.